FWCore/Framework/src/SubProcess.cc

```diff
@@ -744,6 +799,24 @@ namespace edm {
     }
   }
 
+  bool SubProcess::parentProducedProductIsKept(Principal const& parentPrincipal, Principal& principal) const {
+    SelectedProducts const& keptVector = keptProducts()[InProcess];
+    for (auto const& item : keptVector) {
```

Comment:
In principle this looks like the loop could be replaced with `std::find_if()`, but I'm not sure if the resulting code would be more clear or less clear.

Reply:
I tried it (see below). It looks less readable to me so I'm mildly inclined to leave this as is.

Although it does use make use of a standard algorithm. It not a big difference either way and both should work fine. Tests pass with it. If you want it let me know, and I will put it in.

```
  bool SubProcess::parentProducedProductIsKept(Principal const& parentPrincipal, Principal& principal) const {
    SelectedProducts const& keptVector = keptProducts()[InProcess];
    auto const& found = std::find_if(keptVector.begin(), keptVector.end(), [&parentPrincipal, &principal](auto const& item) {
      BranchDescription const& desc = *item.first;
      assert(desc.branchType() == InProcess);
      ProductResolverBase const* parentProductResolver = parentPrincipal.getProductResolver(desc.branchID());
      if (parentProductResolver != nullptr) {
        ProductResolverBase* productResolver = principal.getModifiableProductResolver(desc.branchID());
        if (productResolver != nullptr) {
          return parentProductResolver->branchDescription().produced();
        }
      }
      return false;
    });
    return found != keptVector.end();
  }
```

Reply:
> I tried it (see below). It looks less readable to me so I'm mildly inclined to leave this as is.

Thanks for trying out, I tend to agree with leaving it as it is.